src/test.zig

```diff
@@ -868,10 +868,9 @@ pub const TestContext = struct {
                                 std.testing.zig_exe_path,
                                 "run",
                                 "-cflags",
-                                "-std=c89",
+                                "-std=c99",
```

Comment:
Choice of C89 was deliberate - can we talk about what it might look like to support C89 here before giving up?

Another option we have is to make some exceptions (as you see below with `-Wno-declaration-after-statement`) which act as "TODO"s but still will allow us to notice in other places where we violate C89.

Reply:
Currently the only other c99 feature we are using is compound literals for the optionals and error unions, and I doubt any compiler stuck in c89 will support them. Getting rid of them would require either every call to `renderValue` to pass a temporary value result location or it would require `renderValue` to be able to create declarations. The first option would probably not even work with stuff like function calls and arrays but the second one should be fine.

Reply:
OK, I'm fine with C99, at least for now. I'd much rather prioritize getting the backend done with C99 than imposing possibly unnecessary restrictions on ourselves and impeding progress.